### ISSUE TITLE
Missing UID property for VEVENT

### ISSUE THREAD

**Issue by gabsoftware** [nwcell/ics.js#37]
Each VEVENT should have a unique UID property. Currently they do not have one.

**Comment by nwcell** [comment]
Is this breaking behavior?  I'm marking this as an enhancement otherwise.

**Comment by gabsoftware** [comment]
Yes on my different calendar software it only import the first event and ignore all the others, and if I remember well on Outlook same thing happened.

**Comment by nwcell** [comment]
Yeah, I remember writing this bit.  I'm incrementing the uid by 1 for each event in a multi event download.  I can see there being an issue if you want to merge events into a calendar.

Does anyone know if it's safe to throw a uuid in there?

**Comment by philipp** [comment]
It would be nice to be able to be able to explicitly set an ID or something for it.
The UID is used to identify cancelations and updates to existing events in a calendar.

If you're in the mood for adding at least one optional parameter to "eddEvent", it's probably reasonable to default it to a UUID, but let users override it in case they have a meaningful ID for this particular event.

**Comment by ThirdPrice** [comment]
Has this been implemented? I know it's been a while, but I think this is making an issue with my project as when importing a .ics into calendars, it seems to override the previous import

**Comment by ThisIsJJones** [comment]
I was also having this issue. Went through the code and found the "uidDomain" parameter when initializing the ics. I set this to a random number and the calendar now knows it's a different event. Hope this helps.
